pkg/apis/application/v1alpha1/types.go

```diff
@@ -698,14 +698,20 @@ func (o SyncOptions) HasOption(option string) bool {
 	return false
 }
 
+type CreateNamespaceMetadata struct {
+	Labels      map[string]string `json:"labels,omitempty" protobuf:"bytes,1,opt,name=labels"`
+	Annotations map[string]string `json:"annotations,omitempty" protobuf:"bytes,2,opt,name=annotations"`
```

Comment:
Maybe better providing the name here as well? 
Or even make this struct just have a [ObjectMeta](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.24/#objectmeta-v1-meta) reference in it?

something like:
```golang
type CreateNamespaceMetadata struct {
	metav1.ObjectMeta `json:"metadata"`
```

Reply:
The name is inferred from the resource referencing the to-be-created namespace.